### PR TITLE
feat: extend pack details view with voucher

### PR DESCRIPTION
**Pull request by ViacheslavMylostyvyi**
Asana ticket: https://app.asana.com/0/1201360309822272/120136506981720
Description: Extend pack view with adding voucher info
Знімок екрана  о 13 01 18
4

### CODE REVIEW

File: js/packages/web/src/views/pack/components/PackSidebar/index.tsx [hunk 1]
```diff
@@ -1,32 +1,62 @@
...
     <div className="pack-view__info">
       <div className="info-item">
-        <p>PACK OF</p>
-        <p className="info-count">{pack?.info?.packCards || 0} NFTs</p>
+        <div className="info-item__title">PACK OF</div>
+        <div className="info-item__value">{pack?.info?.packCards || 0} NFTs</div>
+      </div>
+      <div className="info-item">
+        <div className="info-item__title">Royalties</div>
+        <div className="info-item__value">{((art.seller_fee_basis_points || 0) / 100).toFixed(2)}%</div>
```
Comment (b2kdaman) [inline js/packages/web/src/views/pack/components/PackSidebar/index.tsx:47]: Move to separate function

File: js/packages/web/src/views/pack/components/PackSidebar/index.tsx [hunk 2]
```diff
@@ -1,32 +1,62 @@
...
-const PackSidebar = ({ pack }: { pack?: ParsedAccount<PackSet> }) => (
+const PackSidebar = ({ pack, art, id = '' }: { pack?: ParsedAccount<PackSet>, art: Art, id?: string }) => (
   <div className="pack-view__sidebar">
-    <h4>{pack?.info?.name || <Skeleton title={{ width: 300 }} />}</h4>
+    <div className="pack-view__owner">
+      <div className="item-title">Owner</div>
+      {(art.creators || []).map((creator) =>
+        <div key={creator.address}>
+          <MetaAvatar creators={[creator]} size={32} />
+          &emsp;
```
Comment (b2kdaman) [inline js/packages/web/src/views/pack/components/PackSidebar/index.tsx:18]: If you want to use longer space `{'  '}` makes more sense

Comment (b2kdaman) [reply js/packages/web/src/views/pack/components/PackSidebar/index.tsx:18]: Ofc it's better to add additional class to `MetaAvatar` or update `item-name`

File: js/packages/web/src/views/pack/index.tsx [hunk 3]
```diff
@@ -8,18 +8,22 @@ import { useMeta } from '@oyster/common';
...
     useParams();
-  const { packs } = useMeta();
+  const { packs, metadata } = useMeta();
   const { isLoading, mockBlocks, packMetadata, handleFetch } = usePackState(
     packId,
     editionId,
   );
 
   const pack = packs[packId];
+  const metaInfo = metadata.find(meta => meta.info.edition === editionId);
```
Comment (b2kdaman) [inline js/packages/web/src/views/pack/index.tsx:24]: Maybe `meta?.info?.edition`?

File: js/packages/web/src/views/pack/components/PackSidebar/index.tsx [hunk 4]
```diff
@@ -1,32 +1,62 @@
...
+            allowMeshRender={true}
+            artView={true}
+          />
+        ) : (
+          <Skeleton.Image />
+        )
+      }
+
+    </div>
+    <h4 className="pack-view__name">{pack?.info?.name || <Skeleton.Button />}</h4>
```
Comment (m-sebastiyan) [inline js/packages/web/src/views/pack/components/PackSidebar/index.tsx:39]: I think it should be <Skeleton.Paragraph />

File: js/packages/web/src/views/pack/index.tsx (hunk 3 above)
Comment (m-sebastiyan) [reply js/packages/web/src/views/pack/index.tsx:24]: useMemo here pls

File: js/packages/web/src/views/pack/index.tsx [hunk 5]
```diff
@@ -8,18 +8,22 @@ import { useMeta } from '@oyster/common';
...
+  const art = useArt(metaInfo?.pubkey);
```
Comment (m-sebastiyan) [inline js/packages/web/src/views/pack/index.tsx:25]: Why don't you use this hook inside of PackSidebar?
You already provide metaInfo?.pubkey as prop there.

**Comment by m-sebastiyan** [comment]
Add a description to this PR please, 
1. Link to Asana ticket
2. Screenshot
3. Task details

**Review by m-sebastiyan — APPROVED** [review]

File: js/packages/common/src/utils/utils.ts [hunk 6]
```diff
@@ -306,3 +306,7 @@ export function convert(
 export function sleep(ms: number): Promise<void> {
   return new Promise(resolve => setTimeout(resolve, ms));
 }
+
+export function royalty(value: number | undefined): string {
+  return `${((value || 0) / 100).toFixed(2)}%`;
```
Comment (kurpav) [inline js/packages/common/src/utils/utils.ts:311]:
```suggestion
export function royalty(value = 0): string {
  return `${(value / 100).toFixed(2)}%`;
```

**Pull request merged**